// @spectrum-web-components/action-menu/src/ActionMenu.dev.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __decorateClass = (decorators, target, key, kind) => {
  var result = kind > 1 ? void 0 : kind ? __getOwnPropDesc(target, key) : target;
  for (var i = decorators.length - 1, decorator; i >= 0; i--)
    if (decorator = decorators[i])
      result = (kind ? decorator(target, key, result) : decorator(result)) || result;
  if (kind && result)
    __defProp(target, key, result);
  return result;
};
import {
  html
} from "@spectrum-web-components/base";
import { ifDefined } from "@spectrum-web-components/base/src/directives.js";
import { property } from "@spectrum-web-components/base/src/decorators.js";
import { PickerBase } from "@spectrum-web-components/picker";
import "@spectrum-web-components/action-button/sp-action-button.js";
import { ObserveSlotText } from "@spectrum-web-components/shared/src/observe-slot-text.js";
import "@spectrum-web-components/icons-workflow/icons/sp-icon-more.js";
import actionMenuStyles from "./action-menu.css.js";
export class ActionMenu extends ObserveSlotText(PickerBase, "label") {
  constructor() {
    super(...arguments);
    this.selects = void 0;
    this.listRole = "menu";
    this.itemRole = "menuitem";
  }
  static get styles() {
    return [actionMenuStyles];
  }
  get hasLabel() {
    return this.slotHasContent;
  }
  get buttonContent() {
    return [
      html`
                <slot name="icon" slot="icon" ?icon-only=${!this.hasLabel}>
                    <sp-icon-more class="icon"></sp-icon-more>
                </slot>
                <slot name="label" ?hidden=${!this.hasLabel}></slot>
                <slot name="tooltip"></slot>
            `
    ];
  }
  render() {
    return html`
            <sp-action-button
                ?quiet=${this.quiet}
                ?selected=${this.open}
                aria-haspopup="true"
                aria-controls=${ifDefined(this.open ? "menu" : void 0)}
                aria-expanded=${this.open ? "true" : "false"}
                aria-label=${ifDefined(this.label || void 0)}
                id="button"
                class="button"
                size=${this.size}
                @blur=${this.handleButtonBlur}
                @click=${this.handleButtonClick}
                @keydown=${{
      handleEvent: this.handleEnterKeydown,
      capture: true
    }}
                ?disabled=${this.disabled}
            >
                ${this.buttonContent}
            </sp-action-button>
            ${this.renderMenu}
        `;
  }
  update(changedProperties) {
    if (changedProperties.has("invalid")) {
      this.invalid = false;
    }
    super.update(changedProperties);
  }
}
__decorateClass([
  property({ type: String })
], ActionMenu.prototype, "selects", 2);
